src/components/PostDetails/PostDetails.jsx

```diff
@@ -1,45 +1,124 @@
+
+export const PostDetails = ({ selectedPostId }) => {
+  const [selectedPost, setSelectedPost] = useState(null);
+  const [comments, setComments] = useState(null);
+  const [commentsHidden, setCommentsHidden] = useState(false);
+
+  useEffect(() => {
+    getPostDetails(selectedPostId)
+      .then(result => setSelectedPost(result));
+
+    setComments(null);
+
+    getPostComments(selectedPostId)
+      .then(result => setComments(result));
+  }, [selectedPostId]);
+
+  const removeComment = (id) => {
+    deleteComment(id)
+      .then(() => {
+        setComments(currentComment => (
+          currentComment.filter(comment => comment.id !== id)
+        ));
+      });
+  };
+
+  const addNewComment = (name, email, body) => {
+    const commentBody = {
+      name,
+      email,
+      body,
+      postId: selectedPostId,
+    };
+
+    addComment(commentBody)
+      .then((result) => {
+        setComments(currentComment => [...currentComment, result]);
+      });
+  };
+
+  return (
+    <div className="PostDetails">
+      <h2>Post details:</h2>
+
+      {selectedPost ? (
+        <>
+          <section className="PostDetails__post">
+            <p>{selectedPost.title}</p>
+          </section>
+
+          {comments ? (
+            <>
+              <section className="PostDetails__comments">
+                {comments.length === 0
+                  ? 'No comments yet'
+                  : (
+                    <button
+                      type="button"
+                      className="button"
+                      onClick={() => {
+                        setCommentsHidden(currentStatus => !currentStatus);
+                      }}
+                    >
+                      {commentsHidden
+                        ? `Show ${comments.length} comments`
+                        : `Hide ${comments.length} comments`
+                      }
+                    </button>
+                  )}
+
+                {!commentsHidden && (
+                  <ul className="PostDetails__list">
+                    {comments.map(comment => (
+                      <li className="PostDetails__list-item" key={comment.id}>
+                        <button
+                          type="button"
+                          className="PostDetails__remove-button button"
+                          onClick={() => {
+                            removeComment(comment.id);
+                          }}
+                        >
+                          X
+                        </button>
+                        <p>{comment.name}</p>
+                      </li>
+                    ))}
+                  </ul>
+                )}
+              </section>
+
+              <section>
+                <div className="PostDetails__form-wrapper">
+                  <NewCommentForm onAdd={addNewComment} />
+                </div>
+              </section>
+            </>
+          ) : (
+            <Loader />
```

Comment:
To avoid such nesting you can use conditional rendering with loader:
```
const isLoaderVisible = !selectedPost || !comments'
{isLoaderVisible && <Loader />}
```